external-crates/move/crates/move-compiler/src/naming/translate.rs

```diff
@@ -2512,7 +2515,30 @@ fn exp(context: &mut Context, e: Box<E::Exp>) -> Box<N::Exp> {
                             .env
                             .add_diag(diag!(TypeSafety::InvalidCallTarget, (mloc, msg)));
                     }
-                    NE::VarCall(v, nes)
+                    // If this variable refers to a local (num > 0) or it isn't syntax, error.
+                    if !v.value.is_syntax_identifier() {
+                        let msg = format!(
+                            "Unexpected invocation of parameter or local '{}'. \
+                                     Non-syntax variables cannot be invoked as functions.",
```

Comment:
night be helpful to explain what this is. Maybe something like 
"Only a macro's syntax variables, e.g. '$f', can be invoked as functions."